src/plugins/data/common/constants.ts

```diff
@@ -30,6 +30,10 @@
 
 export const DEFAULT_QUERY_LANGUAGE = 'kuery';
 
+// to resolve: copies constants from src/plugins/data/public/data_sources/constants.tsx, should it be removed from there?
+export const S3_GLUE_DATA_SOURCE_TYPE = 's3glue';
```

Comment:
nit: i understand unrelated to this PR is s3glue the only planned polling type of data source?

perhaps we can iteratively improve the existing hardcode resources that has registration which can registration one time within a secondary plugin.

similar to other patterns: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/search/search_service.ts#L112

like certain patterns will utilize a record and if the data source type exists we don't re-register.